lib/ace/plugin_cache.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+require 'puppet/configurer'
+require 'concurrent'
+
+module ACE
+  class PluginCache
+    attr_reader :config, :cache_dir_mutex, :uri
+    def initialize(config)
+      @config = config
+      @uri = URI.parse(config['file-server-uri'])
```

Comment:
cc @da-ar this will interact with you in-flight change for `file-server-uri` -> `puppetserver-uri` renaming.